assets/js/blocks/cart-checkout/checkout/block.js

```diff
@@ -126,10 +114,22 @@ const Checkout = ( {
 	);
 	useEffect( () => {
 		if ( shippingAsBilling ) {
-			setBillingData( shippingAddress );
+			setBillingData( { ...shippingAddress, shippingAsBilling } );
+		} else {
+			setBillingData( { shippingAsBilling } );
 		}
 	}, [ shippingAsBilling, setBillingData ] );
 
```

Comment:
what was the point of adding shippingAsBilling to shipping data? It’s not paired with the backend and I don’t see it being used anywhere else

Reply:
It's used in the example above. I thought it would be useful for extensions to know if the shipping and billing addresses are the same. That would also allow them to display the errors messages in the correct fields.